crates/service/src/prelude/global/i8_cos.rs

```diff
@@ -0,0 +1,266 @@
+use super::G;
+use crate::prelude::scalar::I8;
+use crate::prelude::*;
+
+#[derive(Debug, Clone, Copy)]
+pub enum I8Cos {}
+
+impl G for I8Cos {
+    type Scalar = I8;
+
+    const DISTANCE: Distance = Distance::Cos;
+
+    type L2 = I8L2;
+
+    fn distance(lhs: &[I8], rhs: &[I8]) -> F32 {
+        F32(1.0) - cosine(lhs, rhs)
+    }
+
+    fn elkan_k_means_normalize(vector: &mut [I8]) {
+        l2_normalize(vector)
+    }
+
+    fn elkan_k_means_distance(lhs: &[I8], rhs: &[I8]) -> F32 {
+        super::i8_dot::dot(lhs, rhs).acos()
+    }
+
+    #[multiversion::multiversion(targets(
+        "x86_64/x86-64-v4",
+        "x86_64/x86-64-v3",
+        "x86_64/x86-64-v2",
+        "aarch64+neon"
+    ))]
+    fn scalar_quantization_distance(
```

Comment:
You can raise error for pq/sq on int8. Ref: https://github.com/tensorchord/pgvecto.rs/pull/299/files#diff-230ae43188525a447d40756297495a46c2a21dd29b5913d21056c5296232c5d4R80-R88